core/identity-hub-core/src/main/java/org/eclipse/edc/identityhub/core/services/verifiablepresentation/generators/LdpPresentationGenerator.java

```diff
@@ -159,6 +159,10 @@ private JsonArray toJsonArray(List<VerifiableCredentialContainer> credentials) {
     }
 
     private JsonObject signPresentation(JsonObject presentationObject, SignatureSuite suite, String suiteIdentifier, PrivateKey pk, String publicKeyId, String controller) {
+        if (!publicKeyId.startsWith(controller)) {
```

Comment:
please do not re-use and mutate input arguments. instead, assign a new variable inside the method. For reference, look at `JwtPresentationGenerator`.